RF24_config.h

```diff
@@ -25,6 +25,10 @@
 //#define SPI_UART    // Requires library from https://github.com/TMRh20/Sketches/tree/master/SPI_UART
 //#define SOFTSPI     // Requires library from https://github.com/greiman/DigitalIO
     
+#if !defined(RF24_POWERUP_DELAY_MS)
+#define RF24_POWERUP_DELAY_MS	5
```

Comment:
for documenting purposes, this line should have a preceding doxygen style comment explaining why it exists.
I would use 
```c++
/** User access to internally used delay time (in microseconds) during RF24::powerUp() */
#define RF24_PWR_UP_DELAY 5000
```
I'm not asking to adjust the name of the macro (its just an example). Doxygen will automatically link to `RF24::powerUp()` as long as there's a extra `*` at the beginning of the comment.

Reply:
we could also add a `@see @ref RF24_PWR_UP_DELAY` line in the docs for `powerUp()` to draw attention to this feature.
Maybe a `@warning` paragraph would also be appropriate (due to discussion in #621) in RF24_config.h now that I'm thinking more about it. I'll edit the previous comment example.

Reply:
OK I've switched to ```delayMicroseconds()``` and added the first doc-comment.

Not sure how the ```txDelay``` would work, it seems to be initialized in a rather convoluted manner in ```setDataRate()```?

I'm also willing to call-out the other magic numbers in the code, if you name them for me? Even if a user never changes them, it's good documentation.

Reply:
Could still use that warning to tell users that "here be dragons" with counterfeits.

I was thinking leave `setDataRate()` alone and RF24_TX_DELAY` undefined. Instead, I would just wrap the calls to `delaMicroseconds(txDelay)` with an `#if !defined(RF24_TX_DELAY)` & `#else delayMicroseconds(RF24_TX_DELAY); #endif`. It would be a bit more work to wrap all definitions/declaration of `txDelay` with the same `#ifndef` soup, but maybe it would be required if the compiler warns that `txDelay` isn't used when RF24_TX_DELAY is defined.

TBH these are the only 2 delay times that could be adjustable (if that's what you mean by "magic numbers"), and `txDelay` is a little too advanced for documenting it (would require a direct link to the datasheet about transmission time) -- not even sure what part of the docs would be appropriate for explaining a macro that's left undefined (maybe in `stopListening()`). I'd defer to [my last comment from #683 for more details on other delays](https://github.com/nRF24/RF24/issues/683#issuecomment-729999633).

Reply:
On second thought, it would probably be a lot easier to expose the `txDelay` to the public API instead of keeping it private and using a macro. Although, I'd like permission from @TMRh20 before making any commits about `txDelay`.

Reply:
## I'm an idiot! 
`txDelay` is already exposed to the public API & [it is documented already](https://tmrh20.github.io/RF24/classRF24.html#a958fb99f54415101ca008ab11b3bfe79). Please disregard everything I said about `txDelay`, seriously I'm an idiot sometimes.

@jscrane just a warning in RF24_POWERUP_DELAY & this PR is good to go. I can't think of any other delay that deserves any/more documentation than is already given.

Reply:
Well by "magic numbers" I meant raw numeric constant arguments to delay() and delayMicroseconds().

Some examples:
```
    #if !defined(F_CPU) || F_CPU > 20000000
    delayMicroseconds(10);
    #endif
```
Should the following delay in RF24::begin() also be RF24_POWERUP_DELAY or is it different?

```
    // Must allow the radio time to settle else configuration bits will not necessarily stick.
    // This is actually only required following power up but some settling time also appears to
    // be required after resets too. For full coverage, we'll always assume the worst.
    // Enabling 16b CRC is by far the most obvious case if the wrong timing is used - or skipped.
    // Technically we require 4.5ms + 14us as a worst case. We'll just call it 5ms for good measure.
    // WARNING: Delay is based on P-variant whereby non-P *may* require different timing.
    delay(5);
```

```
    write_register(EN_RXADDR, read_register(EN_RXADDR) | _BV(pgm_read_byte(&child_pipe_enable[0]))); // Enable RX on pipe0

    //delayMicroseconds(100);
```

I have updated the doc-comment.

Reply:
>
    #if !defined(F_CPU) || F_CPU > 20000000
    delayMicroseconds(10);
    #endif

This number cannot change and is actually mentioned in the one of the source comments. It is the required pulse time for using 1 payload from the TX FIFO. However, the `#ifdef` is needed to compensate for more speedy MCUs (i.e. ATMega328 runs slow enough to not need this call). `write()` (cleverly) gets around this delay by waiting for the radio to assert an IRQ flag... I wouldn't worry about documenting this one because my changes to the docs about `reUseTX()` already mention this delay (PR coming soon), and all other `write*()` or `txStandBy()` functions abstract the need to understand this delay from a user perspective.

>
    // Must allow the radio time to settle else configuration bits will not necessarily stick.
    // This is actually only required following power up but some settling time also appears to
    // be required after resets too. For full coverage, we'll always assume the worst.
    // Enabling 16b CRC is by far the most obvious case if the wrong timing is used - or skipped.
    // Technically we require 4.5ms + 14us as a worst case. We'll just call it 5ms for good measure.
    // WARNING: Delay is based on P-variant whereby non-P *may* require different timing.
    delay(5);

- I have no idea what this comment relates to. I have a feeling its a one-time delay that happens to use the same amount of time (as `powerUp()`), but I would leave it as is because it feels like the long preceding comment is speaking from experience rather than datasheet guidance.
- Not sure if this is relevant, but I vaguely remember the datasheet saying that ["the nRF24L01+ must be in a standby or power down mode before writing to the configuration registers"](https://tmrh20.github.io/RF24/tmrh20/nRF24L01P_datasheet_v1.pdf#G1105319). After abandoning all attempts to appease that stipulation, I have not seen any problems arise from ignoring it. 

>
    write_register(EN_RXADDR, read_register(EN_RXADDR) | _BV(pgm_read_byte(&child_pipe_enable[0]))); // Enable RX on pipe0
    //delayMicroseconds(100);

IIRC, this code is from `stopListening()`. I assumed the commented delay is replaced with the `txDelay` usage.

Reply:
I also went through all the delays ([in this comment](https://github.com/nRF24/RF24/issues/683#issuecomment-729999633)) and found only the `powerUp()` delay that could use circumstantial treatment.